app/controllers/github_repos_controller.rb

```diff
@@ -38,13 +38,18 @@ def update_or_create
 
   private
 
-  def fetch_repositories_from_github(known_repositories_ids)
+  def fetch_repositories_from_github(known_repositories)
     client = Github::OauthClient.for_user(current_user)
-
-    client.repositories(visibility: :public).map do |repo|
-      repo.featured = known_repositories_ids.include?(repo.id)
+    repos = client.repositories(visibility: :public).map do |repo|
+      repo.featured = known_repositories.delete_if { |known| known.github_id_code == repo.id }.present?
       repo
-    end.sort_by(&:name)
+    end
+
+    # Remove pinned repositories that were removed from GH or are now private,
+    # since the user will not be able to remove them by themselves.
+    known_repositories.each(&:destroy)
```

Comment:
Since we're using `delete_if` above, this will only contain the repositories we still consider featured but that are no longer part of the user's repos with public visibility according to GH.